src/dbnode/storage/block/merged_block_reader.go

```diff
@@ -129,8 +129,18 @@ func (r *dbMergedBlockReader) mergedReader() (xio.BlockReader, error) {
 		r.readers[i] = nil
 	}
 
+	var (
+		stream xio.SegmentReader
+		ok     bool
+	)
+	stream, ok = r.encoder.Stream(encoding.StreamOpts{})
+	if !ok {
+		// xio.BlockReader expects SegmentReader to be nil if there
+		// is no data.
+		stream = nil
```

Comment:
Hm, this seems counter-intuitive, instead of now possibly hitting a nil pointer case because we didn't have the `ok` var we now may do the wrong thing by default if it actually has no data.

Can we perhaps just make BlockReader do the same thing it used to do with a nil segmentreader before?

We could un-export all the fields of `BlockReader`, create a constructor, then basically inside the call to `SegmentReader()` on `BlockReader` enforce a nil return if it's instead a stream that's empty.

Or we can do other things to force the constructs at the type level rather than have to be careful when constructing one, like we're having to do here.

Reply:
Yeah sure, this was clearly me just being lazy after working on this P.R for awhile. I like the constructor approach, I’ll give that a shot

Reply:
Let me know how it goes, apologies that `BlockReader` doesn't already have a constructor =[